compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -165,6 +165,8 @@ pub struct Elaborator<'context> {
     /// Each element of the Vec represents a scope with every scope together making
     /// up all currently visible definitions. The first scope is always the global scope.
     comptime_scopes: Vec<HashMap<DefinitionId, comptime::Value>>,
+
+    unresolved_globals: VecDeque<UnresolvedGlobal>,
```

Comment:
Why a `VecDeque` vs. e.g. a `BTreeMap<GlobalId, UnresolvedGlobal>`? It appears that most access is by `GlobalId`

Reply:
I had a map as well at first but wanted to preserve the ordering from the input Vec since the globals there are in definition order